src/netbase.h

```diff
@@ -246,4 +246,17 @@ void InterruptSocks5(bool interrupt);
  */
 bool Socks5(const std::string& strDest, uint16_t port, const ProxyCredentials* auth, const Sock& socket);
 
+/**
+ * Determine if a port is "bad" from the perspective of attempting to connect
+ * to a node on that port.
+ * See
+ * https://fetch.spec.whatwg.org/#port-blocking
+ * https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/net/base/port_util.cc
+ * https://hg.mozilla.org/mozilla-central/file/tip/netwerk/base/nsIOService.cpp
+ * https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736
```

Comment:
Still not sure about few things and comment above was marked as resolved even though few other reviewers have similar confusion about ports:

1. What is a BAD port for Bitcoin Core?
2. Why browsers (bad ports) are referenced here because Bitcoin Core is not a browser and I was unable to find anything relevant for other P2P networks. Example: [i2p](https://geti2p.net/en/faq#ports)
3. https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736 this doesn't look like a big issue or maybe there is some misunderstanding
4. What was the reason to allow p2p connections to nodes that are not using default ports? Why would someone use non-default port? Two reasons that I could think of: Privacy and Security. If someone wants to use a port that looks like other service but is Bitcoin Core, isn't that something we should encourage/allow?

Reply:
> ... comment above was marked as resolved even though ...

Which comment? https://github.com/bitcoin/bitcoin/pull/23542#discussion_r752571046 is still open.

> 1. What is a BAD port for Bitcoin Core?

Since this PR, if it gets merged, a bad port (for making automatic outgoing connections to) is one for which `IsBadPort()` returns `true`.

> 2. Why browsers (bad ports) are referenced here because Bitcoin Core is not a browser ...

https://github.com/bitcoin/bitcoin/pull/23542#discussion_r753020571

> 4. What was the reason to allow p2p connections to nodes that are not using default ports?

From this PR OP: "because connections over clearnet that involve
port 8333 make it easy to detect, analyze, block or divert Bitcoin p2p
traffic before the connection is even established (at TCP SYN time)"

> Why would someone use non-default port?

See above, and also the reasons you state below:

> Two reasons that I could think of: Privacy and Security. If someone wants to use a port that looks like other service but is Bitcoin Core, isn't that something we should encourage/allow?

Yes. This is what the current PR aims to achieve.

Reply:
Replied here: https://github.com/bitcoin/bitcoin/pull/23542#discussion_r766906987